### PR TITLE
Support VF self Management

### PR DESCRIPTION
**Pull request by hustcat**
Support VF self Management. Copy from @rkamudhan 's [commit](https://github.com/Intel-Corp/sriov-cni/commit/efba06fb76e7598f3a394e8cfc01ab330520d4ea).

### CODE REVIEW

**Comment by hustcat** [comment]
@rkamudhan I'v made changes from your commit to be compatiable with master branch. Please take a look.

**Comment by rkamudhan** [comment]
@hustcat I will check the plugin, and let you know by tomorrow.

File: sriov/sriov.go [hunk 1]
```diff
@@ -49,36 +51,31 @@ func loadConf(bytes []byte, args string) (*SriovConf, error) {
...
 	}
 
-	vfDir := fmt.Sprintf("/sys/class/net/%s/device/virtfn%d/net", masterName, vfIdx)
-	if _, err := os.Lstat(vfDir); err != nil {
-		return err
-	}
-
-	infos, err := ioutil.ReadDir(vfDir)
+	// VF device name
+	vfDevName, err := getVFDeviceName(masterName, vfIdx)
```
Comment (rkamudhan) [inline sriov/sriov.go]: Do we need to call the getVFDeviceName() twice inside the allocFreeVF() function ?

**Review by rkamudhan — CHANGES_REQUESTED** [review]
With reference to commit https://github.com/Intel-Corp/sriov-cni/commit/efba06fb76e7598f3a394e8cfc01ab330520d4ea#diff-8becae7a8222be03e1bcd86d9adaa117R115

File: sriov/sriov.go (hunk 1 above)
Comment (hustcat) [reply sriov/sriov.go]: Maybe return device name in `allocFreeVF` is better. what do you think? @rkamudhan

**Comment by hustcat** [comment]
@rkamudhan Updated, PTAL:)

**Comment by hustcat** [comment]
@rkamudhan Have you still care this PR?

**Comment by rkamudhan** [comment]
@hustcat I will review it, in other 2 days and confirm you.

**Comment by rkamudhan** [comment]
@hustcat merged the code from the Intel repo [https://github.com/Intel-Corp/sriov-cni/commit/efba06fb76e7598f3a394e8cfc01ab330520d4ea](https://github.com/Intel-Corp/sriov-cni/commit/efba06fb76e7598f3a394e8cfc01ab330520d4ea)

**Pull request merged**